conan_provider.cmake

```diff
@@ -66,15 +66,22 @@ function(detect_arch ARCH)
             message(WARNING "CMake-Conan: Multiple architectures detected, this will only work if Conan recipe(s) produce fat binaries.")
         endif()
     endif()
-    if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64|arm64" OR CMAKE_OSX_ARCHITECTURES MATCHES arm64)
+    if(CMAKE_SYSTEM_NAME MATCHES "Darwin|iOS|tvOS|watchOS")
+        set(host_arch ${CMAKE_OSX_ARCHITECTURES})
+    elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
+        set(host_arch ${CMAKE_CXX_COMPILER_ARCHITECTURE_ID})
+    else()
+        set(host_arch ${CMAKE_SYSTEM_PROCESSOR})
+    endif()
+    if(host_arch MATCHES "aarch64|arm64|ARM64")
         set(_ARCH armv8)
-    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "armv7-a|armv7l" OR CMAKE_OSX_ARCHITECTURES MATCHES armv7)
+    elseif(host_arch MATCHES "armv7|armv7-a|armv7l|ARMV7")
         set(_ARCH armv7)
-    elseif(CMAKE_OSX_ARCHITECTURES MATCHES armv7s)
+    elseif(host_arch MATCHES armv7s)
         set(_ARCH armv7s)
-    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i686" OR CMAKE_OSX_ARCHITECTURES MATCHES i386)
+    elseif(host_arch MATCHES "i686|i386|X86")
         set(_ARCH x86)
-    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "AMD64|amd64|x86_64" OR CMAKE_OSX_ARCHITECTURES MATCHES x86_64)
+    elseif(host_arch MATCHES "amd64|x86_64|x64")
```

Comment:
why is `AMD64` being removed? is it the case that `MATCHES` will match them regardless of lower-case/upper-case?

Reply:
`CMAKE_CXX_COMPILER_ARCHITECTURE_ID` lists the arch as `x64`, but since `CMAKE_SYSTEM_PROCESSOR` could still be used for non-MSVC compilers on Windows, I put it back.